frontend/src/static/js/v_summary.js

```diff
@@ -554,7 +554,9 @@ window.vSummary = {
         };
 
         this.addLineContributionWeek(endOfWeekMsWithinUntilMs, week, commits);
-        res.push(week);
+        if (week.commitResults.length > 0) {
+          res.push(week);
```

Comment:
is this change still necessary?

Reply:
The reason why I added this is because currently, weeks' with zero commits are still added to the result (`res`). This change removes the weeks with zero commits as they are not useful.

Reply:
It is preferrable to have only 1 logic governs 1 behavior, either `if (week.commitResults.length > 0) {`  or  `.zoom__day(v-if="day.insertions > 0")`, not both. 
Reason: 
* Its duplicated / unnecessary logic
* The next dev (if it's not you) who is going to implement a change on this, going to waste some precious amount of time wondering about the weird behavior before realizing there's a second logic preventing the change

Reply:
Yes I understand, but the logic is not really duplicated though.

My reasoning for implementing `if (week.commitResults.length > 0) {` is because there maybe weeks with no commits at all, so by having this condition we will not have empty weeks. By not having this condition will lead to unnecessary `<!---->` in the DOM:
![Screen Shot 2020-01-23 at 9 41 31 PM](https://user-images.githubusercontent.com/32864116/72989491-4f1b6c80-3e29-11ea-9647-1c0647db4b17.png)

The `.zoom__day(v-if="day.insertions > 0")` is to not include commits with only deletions, as we only calculate the ramp size based on number of insertions. 

But I understand that it may confuse other devs on this. Should I remove the `if (week.commitResults.length > 0) {` logic?

Reply:
>The .zoom__day(v-if="day.insertions > 0") is to not include commits with only deletions, as we only calculate the ramp size based on number of insertions.

Hmm, ideally it should include all form of work done (including deletion).

@damithc should we put up an issue to add in the deletion commits? We can subsequently add a toggle that allows user to hide the deletion commits if its not relevant to them.

Reply:
> @damithc should we put up an issue to add in the deletion commits? We can subsequently add a toggle that allows user to hide the deletion commits if its not relevant to them.

It is a good idea but I'm not sure how deep the ramification go. For example, translating lines deleted into a ramp size will be tricky. Should the ramp size reflect linesAdded + linesDeleted?

Reply:
@eugenepeh in this case, should I remove the check for `.zoom__day(v-if="day.insertions > 0")` and put up an issue to modify the ramp size to reflect linesAdded + linesDeleted?

Reply:
Yes, please do remove the check. The commit view should show all commits including deletion.

For the ramp size, it would be tricky to include the lines deleted as highlighted by prof. I will put up an issue to discuss the modification.

Reply:
Yep done so.